DotaStats/Modules/PlayerInfoModule/View/PlayerInfoModuleViewController.swift

```diff
@@ -0,0 +1,170 @@
+            tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+        ])
+    }
+
+    // MARK: ErrorView Constraints
+
+    private func setupErrorViewConstraints() {
+        view.addSubview(errorView)
+        let errorConstraint = errorView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor)
+        NSLayoutConstraint.activate([
+            errorView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            errorView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            errorConstraint,
+            errorView.heightAnchor.constraint(equalToConstant: 90)
+        ])
+        self.errorConstraint = errorConstraint
+    }
+
+    // MARK: Show/hide errors function
+
+    func showError() {
+        UIView.animate(withDuration: 0.5,
+                       delay: 0.0,
+                       options: [.curveEaseInOut]) {
+            self.errorConstraint?.constant = 35
+            self.errorView.alpha = 1
+            self.view.layoutIfNeeded()
+        }
+    }
+
+    func hideError() {
+        UIView.animate(withDuration: 0.5,
+                       delay: 0.0,
+                       options: [.curveEaseOut]) {
+            self.errorConstraint?.constant = 0
+            self.errorView.alpha = 0
+            self.view.layoutIfNeeded()
+        }
+    }
+
+    @objc func handleTap(_: UITapGestureRecognizer) {
+        hideError()
+    }
+}
+
+extension PlayerInfoModuleViewController: PlayerInfoModuleViewInput {
+    func update(state: PlayerInfoModuleViewState) {
+        switch state {
+        case .loading:
+            hideError()
+            spiner.color = ColorPalette.accent
+            view.addSubview(spiner)
+            spiner.center = view.center
+            spiner.startAnimating()
+        case .error:
+            showError()
+            spiner.removeFromSuperview()
+        case .successWL, .successMain, .successMatch:
+            hideError()
+            spiner.removeFromSuperview()
+            if first {
+                setupTableView()
+                first = false
+            }
+            tableView.reloadData()
+        }
+    }
+}
+
+extension PlayerInfoModuleViewController: UITableViewDelegate, UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        output?.getRowsInSection(section: section) ?? 0
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard
+            let data = output?.getCellData(forRow: indexPath.row),
+            let cell = tableView.dequeueReusableCell(
+                withIdentifier: data.reuseIdentificator,
+                for: indexPath) as? (UITableViewCell & PlayerInfoCellConfigurable)
+        else {
+            // TODO: - Error handling
+            return UITableViewCell()
+        }
```

Comment:
Здесь лучше `cell` получить в начале, потом попытаться скастить и, если не случилось, вернуть ее же
```suggestion
        let cell = tableView.dequeueReusableCell(
                withIdentifier: data.reuseIdentificator,
                for: indexPath)
        guard
            let data = output?.getCellData(forRow: indexPath.row),
            let cell = playerInfoCell as? (UITableViewCell & PlayerInfoCellConfigurable)
        else {
            // TODO: - Error handling
            return cell
        }
```